### PR TITLE
Team page clean up

### PR DESCRIPTION
**Pull request by nawaji**
Revamped the team member cards and implemented the search handler function for the Team page.

### CODE REVIEW

File: frontend/pages/team.js [hunk 1]
```diff
@@ -28,25 +48,20 @@ const Team = () => {
...
+                </div>
+                <div className='col col-sm-4 col-md-3 col-lg-2'>
+                  <label htmlFor='team-filter-select' className='form-label'>Filter by</label>
+                  <select className='form-select' id='team-filter-select' onChange={searchHandler}>
+                    <option value='Default'>None</option>
+                    <option value='Class'>Class</option>
+                    <option value='Degree'>Degree</option>
+                    <option value='Group'>Group</option>
+                  </select>
+                </div>
```
Comment (stevezac-osu) [inline frontend/pages/team.js:64]: I say get rid of (or comment out) this entire section.  It's not currently being implemented, so we should hide this until it is.

File: frontend/models/people.js [hunk 2]
```diff
@@ -12,9 +12,26 @@ function Person (rawData) {
...
   p.degrees = DegreeService.getDegrees({ id__eq: p.degrees })
+  p.RawData.formattedPersonName = p.RawData.FirstName + ' ' + p.RawData.LastName
+  p.RawData.personClass = p.personClass.Name
+
   return p
 }
 
-const PersonService = BaseService.constructDefaultService('api::person.person', 'person', 'people', Person)
+const peopleFilterFunctions = Object.assign({}, BaseService.filterFunctions)
+peopleFilterFunctions.searchNameAndClass = (key, value, obj) => obj.formattedPersonName.includes(value) || obj.FirstName.includes(value) || obj.personClass.includes(value)
```
Comment (stevezac-osu) [inline frontend/models/people.js:22]: One thing we should do here for the searches is to force the search criteria to either upper case or lower case, and the value that we search should be either upper case or lower case (to match what we switched the criteria to).  This way, searching is not case specific.  (note, this need to be changed for the other search section as well.)

Example:

`obj.formattedPersonName.toLowerCase().includes(value.toLowerCase()) || obj.FirstName.toLowerCase().includes(value.toLowerCase()) || obj.personClass.toLowerCase().includes(value.toLowerCase())`

File: frontend/components/Personcard.js [hunk 3]
```diff
@@ -1,22 +1,14 @@
...
-          preserveAspectRatio='xMidYMid slice'
-        />
-      </div>
-      <div className={style.bottom}>
-        <h5><strong>{person.FirstName} {person.LastName}</strong></h5>
-        <hr className={style.horizontalbar} />
-        <h6 className='beaverorange'>{person.personClass.Name}</h6>
+    <div className='col'>
+      <div className='card'>
+        <img src={person.photoImage.url} className='card-img-top' alt='Team member portrait' />
```
Comment (stevezac-osu) [inline frontend/components/Personcard.js:5]: For some reason, I am seeing Patricks Picture showing up a few pixels larger than the other students.  Please look into the CSS for this.

File: frontend/components/Personcard.js [hunk 4]
```diff
@@ -1,22 +1,14 @@
...
+        <div className='card-body'>
+          <h5 className='card-title'>{person.formattedPersonName}</h5>
+          <h6 className='card-subtitle mb-2 beaverorange'>{person.personClass.Name}</h6>
```
Comment (stevezac-osu) [inline frontend/components/Personcard.js:8]: This is not a blocking error, but as a future item, we need to look into the usage of header tags (h1-h6) and decide if they are necessary in that context.

File: frontend/pages/team.js [hunk 5]
```diff
@@ -28,25 +48,20 @@ const Team = () => {
...
-                        <option value='Group'>Group</option>
-                      </select>
-                    </li>
-                  </div>
-                </ul>
+              <div className='row justify-content-between'>
+                <div className='col col-sm-4'>
+                  <label htmlFor='team-search' className='form-label'>Search</label>
+                  <input type='search' className='form-control' id='team-search' onChange={searchHandler} placeholder='Search by name, class, degree, group' />
+                </div>
```
Comment (stevezac-osu) [inline frontend/pages/team.js:55]: If we're getting rid of / commenting out the following section, please remove col-sm-4.

Remember, Bootstrap is a base 12 system meaning that there are always 12 columns.  Make sure that you expand items out as necessary.  Meaning, if the following section has: 
`col col-sm-4 col-md-3 col-lg-2`, then this section should have `col col-sm-8 col-md-9 col-lg-10`

**Review by stevezac-osu — DISMISSED** [review]
Hey There, I have a few minor changes I am requesting.  Please review the comments in the code.

File: frontend/components/Personcard.js (hunk 4 above)
Comment (nawaji) [reply frontend/components/Personcard.js:8]: Good catch, for now I'll edit the cards to follow the same formatting as the project cards. The names will be h5, and the description/sub-text will be paragraph text with the 'text-muted' subclass.

**Review by stevezac-osu — APPROVED** [review]
Looks Good

**Pull request merged**